dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -208,19 +208,28 @@ private long getBackoffNanoseconds() {
         }
 
         ListenableFuture<Response> handleHttpResponse(Response response) {
-            if (Responses.isQosStatus(response)) {
+            if (isRetryableQosStatus(response)) {
                 return incrementFailuresAndMaybeRetry(response, qosThrowable, retryDueToQosResponse);
             }
 
             if (response.code() == 500 && safeToRetry(endpoint.httpMethod())) {
                 return incrementFailuresAndMaybeRetry(response, serverErrorThrowable, retryDueToServerError);
             }
 
-            // TODO(dfox): if people are using 308, we probably need to support it too
-
             return Futures.immediateFuture(response);
         }
 
+        private boolean isRetryableQosStatus(Response response) {
+            if (!Responses.isQosStatus(response)) {
+                // Not QoS
+                return false;
+            }
+            // ServerQos.PROPAGATE_429_and_503_TO_CALLER should only propagate too-many-requests and unavailable.
+            boolean shouldPropagate = shouldPropagate429And503(serverQoS)
+                    && (Responses.isTooManyRequests(response) || Responses.isUnavailable(response));
+            return !shouldPropagate;
+        }
```

Comment:
I think I'd pitch:
```java
        private boolean isRetryableQosStatus(Response response) {
            switch (serverQoS) {
                case AUTOMATIC_RETRY:
                    return Responses.isQosStatus(response);
                case PROPAGATE_429_and_503_TO_CALLER:
                    return Responses.isRetryOther(response);
            }
        }
```

Reply:
I’m worried that if we add new QoS statuses they would be erroneously retried

Reply:
I see what you mean, subtracting the 429 and 503 is a little more defensive... although we have got a long way without adding any more statuses! How about

```java
        private boolean isRetryableQosStatus(Response response) {
            switch (serverQoS) {
                case AUTOMATIC_RETRY:
                    return Responses.isQosStatus(response);
                case PROPAGATE_429_and_503_TO_CALLER:
                    return Responses.isQosStatus(response)
                            && !Responses.isTooManyRequests(response)
                            && !Responses.isUnavailable(response);
            }
        }
```

Reply:
sgtm 👍